File: tests/fixtures/ts_samples/sample.go
package main

import "fmt"

type Animal interface {
	Speak() string
}

type Dog struct {
	Name string
}

type DogAlias = Dog

func (d Dog) Speak() string {
	return "Woof"
}

func Greet(name string) {
	fmt.Println("Hello", name)
}

File: tests/fixtures/ts_samples/sample.ts
interface Greeter {
  greet(name: string): string;
}

type Alias = string;

enum Color {
  Red,
  Green,
  Blue,
}

class Dog implements Greeter {
  greet(name: string): string {
    return "Hello, " + name;
  }
}

function hello(name: string): string {
  return "Hello, " + name;
}
